### ISSUE TITLE
refactor pth_06 to be more object thinking

### ISSUE THREAD

**Issue by eemhu** [teragrep/pth_06#12]
ID<53>
Reporter <@eemhu>

related to spark 3 update

**Comment by eemhu** [comment]
* first getAndIncrementLatestOffset call returns some data, but the next call results in NULL latestOffset- StreamDBClient does not find next hour

 [LatestOffset] Record of next hour with offset 1606590000 was empty!
 [LatestOffset] Within limit, latest offset is now null

**Comment by eemhu** [comment]
[LatestOffset] Set next to 1606546800
 [LatestOffset] Within limit!
 [LatestOffset] Set next to 1606568400
 [LatestOffset] Within limit!
 [LatestOffset] Set next to 1606575600
[LatestOffset] Within limit!
 [LatestOffset] Set next to 1606590000
 [LatestOffset] Within limit!
[LatestOffset] Record of next hour with offset 1606590000 was empty!
 [LatestOffset] Within limit, latest offset is now 1606590000
[LatestOffset] Record of next hour with offset 1606590000 was empty!
 [LatestOffset] Within limit, latest offset is now null

**Comment by kortemik** [comment]
should be quite much more

```
MariaDB [journaldb]> select count(id) from logfile where logtag='alert_examples' and logdate > "2020-11-27";
+-----------+
| count(id) |
+-----------+
|      3967 |
```

**Comment by eemhu** [comment]
manually setting earliest values to be further gets more data, so definitely something wrong with seeking more results.

**Comment by eemhu** [comment]
a lot more data is now being returned, however there are duplicates. "example" index returns a duplicate of every second row.

SliceTable needs to clear the ones already returned so they wont get returned twice.

**Comment by eemhu** [comment]
The duplicates were from kafka. Archive only query returns expected 730 rows, kafka only query returns 365, which are the duplicates.

incrementAndGetLatestOffset still scrolls to the last offset. Need to rethink the logic and make sure it scrolls them in small intervals. For example the current implementation returns example index in 6x 100 and 1x 30 batches. Object thinking version in one batch of 730.

**Comment by StrongestNumber9** [comment]
Is this still relevant? Internal object-refactoring branch was merged with #36 @eemhu @kortemik
